src/components/PopupWithForm.js

```diff
@@ -16,16 +17,23 @@ export default class PopupWithForm extends Popup {
     return inputValues;
   }
 
-  sendInputValues() {
-    return this._getInputValues();
+  _renderSaving(isSaving) {
+    const submitButton = this._popupForm.querySelector(".modal__save");
+    if (isSaving) {
+      submitButton.textContent = "Saving...";
+    } else {
+      submitButton.textContent = this._defaultSubmitButtonText;
+    }
   }
 
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`